File: workflows-java/src/main/java/com/example/flow/FlowThatGetsHospitalized.java
package com.example.flow;

import co.paralleluniverse.fibers.Suspendable;
import net.corda.core.flows.*;
import net.corda.core.utilities.ProgressTracker;

@InitiatingFlow
@StartableByRPC
public class FlowThatGetsHospitalized extends FlowLogic<Void> {
    private final ProgressTracker progressTracker = new ProgressTracker();

    @Override
    public ProgressTracker getProgressTracker() {
        return progressTracker;
    }

    @Suspendable
    @Override
    public Void call() {
        throw new HospitalizeFlowException("Ouch!");
    }
}

File: workflows-kotlin/src/main/kotlin/com/example/flow/FlowThatGetsHospitalized.kt
package com.example.flow

import co.paralleluniverse.fibers.Suspendable
import net.corda.core.flows.FlowLogic
import net.corda.core.flows.HospitalizeFlowException
import net.corda.core.flows.InitiatingFlow
import net.corda.core.flows.StartableByRPC
import net.corda.core.utilities.ProgressTracker

@InitiatingFlow
@StartableByRPC
class FlowThatGetsHospitalized : FlowLogic<Void?>() {
    override val progressTracker = ProgressTracker()

    @Suspendable
    override fun call(): Void {
        throw HospitalizeFlowException("Ouch!")
    }
}